components/camel-http/src/main/java/org/apache/camel/component/http/OAuth2ClientConfigurer.java

```diff
@@ -37,55 +45,143 @@ public class OAuth2ClientConfigurer implements HttpClientConfigurer {
     private final String clientSecret;
     private final String tokenEndpoint;
     private final String scope;
+    private final boolean cacheTokens;
+    private final Long cachedTokensDefaultExpirySeconds;
+    private final Long cachedTokensExpirationMarginSeconds;
+    private final static Map<OAuth2URIAndCredentials, TokenCache> tokenCache = new HashMap<>();
```

Comment:
It is great to see this feature added, but this should be made thread safe (e.g. by using `ConcurrentHashMap`) otherwise different threads may not see each others updates and in the worst case there can be an infinite loop during concurrent modification.

Reply:
yeah thanks